### PR TITLE
Netty axis2 bridge improvements

### PR DESCRIPTION
**Pull request by tharinduwijewardane**
Get axis2 message formatters to format the message.
Change packaging of netty axis2 bridge to bundle.
Clean up code.

### CODE REVIEW

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java [hunk 1]
```diff
@@ -123,4 +131,117 @@ public static boolean isEmptyPayloadStream(InputStream inputStream) throws IOExc
...
+            // If we are doing rest better default to Application/xml formatter
+            if (msgContext.isDoingREST()) {
+                String httpMethod = (String) msgContext.getProperty(Constants.Configuration.HTTP_METHOD);
+                if (Constants.Configuration.HTTP_METHOD_GET.equals(httpMethod) ||
+                    Constants.Configuration.HTTP_METHOD_DELETE.equals(httpMethod)) {
+                    return new XFormURLEncodedFormatter();
+                }
+                return new ApplicationXMLFormatter();
+            } else {
+                // Lets default to SOAP formatter
```
Comment (menuka94) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]:
```suggestion
                // default to SOAP formatter
```

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java [hunk 2]
```diff
@@ -123,4 +131,117 @@ public static boolean isEmptyPayloadStream(InputStream inputStream) throws IOExc
...
+        } else {
+            format = new OMOutputFormat();
+        }
+
+        msgContext.setDoingMTOM(TransportUtils.doWriteMTOM(msgContext));
+        msgContext.setDoingSwA(TransportUtils.doWriteSwA(msgContext));
+        msgContext.setDoingREST(TransportUtils.isDoingREST(msgContext));
+
+        /*
+         *  PassThroughConstants.INVOKED_REST set to true here if isDoingREST is true -
```
Comment (menuka94) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]:
```suggestion
         * PassThroughConstants.INVOKED_REST set to true here if isDoingREST is true -
```

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java [hunk 3]
```diff
@@ -123,4 +131,117 @@ public static boolean isEmptyPayloadStream(InputStream inputStream) throws IOExc
...
+         *  this enables us to check whether the original request to the endpoint was a
```
Comment (menuka94) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]:
```suggestion
         * this enables us to check whether the original request to the endpoint was a
```

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/RequestUtils.java [hunk 4]
```diff
@@ -176,17 +179,17 @@ static String getRestUrlPostfix(String uri, String servicePath) {
...
 
     static boolean isRest(String contentType) {
         return contentType != null &&
-                !contentType.contains(SOAP11Constants.SOAP_11_CONTENT_TYPE) &&
-                !contentType.contains(SOAP12Constants.SOAP_12_CONTENT_TYPE);
+               !contentType.contains(SOAP11Constants.SOAP_11_CONTENT_TYPE) &&
+               !contentType.contains(SOAP12Constants.SOAP_12_CONTENT_TYPE);
     }
 
     static int populateSOAPVersion(MessageContext msgContext, String soapActionHeader, String contentType) {
```
Comment (isudana) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/RequestUtils.java]: what does it mean by having soap version 0?

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java [hunk 5]
```diff
@@ -161,34 +161,38 @@ private void sendForward(MessageContext msgCtx, HttpCarbonMessage httpCarbonMess
...
-            final OutputStream outputStream = outboundMsgDataStreamer.getOutputStream();
-            OMElement omElement = msgCtx.getEnvelope().getBody().getFirstElement();
-            writeToStream(outputStream, omElement.toString());
+            formatAndWrite(msgCtx, httpCarbonMessage);
         }
-
     }
 
-    private void writeToStream(OutputStream outputStream, String omElementString) {
+    private void formatAndWrite(MessageContext msgCtx, HttpCarbonMessage httpCarbonMessage) {
```
Comment (isudana) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: Shall we rename this to writeWithFormatter or something similar, having and gives the idea that this method is doing multiple stuff.

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java [hunk 6]
```diff
@@ -161,34 +161,38 @@ private void sendForward(MessageContext msgCtx, HttpCarbonMessage httpCarbonMess
...
-                LOG.error(BridgeConstants.BRIDGE_LOG_PREFIX + e.getMessage());
+                LOG.error("{} error closing outputStream: {}", BridgeConstants.BRIDGE_LOG_PREFIX,
+                          e.getMessage());
             }
         }
     }
 
     @Override
     public void cleanup(MessageContext messageContext) {
+        LOG.debug("cleanup");
```
Comment (isudana) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: Add a proper log message.

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java [hunk 7]
```diff
@@ -161,34 +161,38 @@ private void sendForward(MessageContext msgCtx, HttpCarbonMessage httpCarbonMess
...
     }
 
     @Override
     public void stop() {
+        LOG.debug("stop");
```
Comment (isudana) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: Add a proper log message.

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java [hunk 8]
```diff
@@ -123,4 +131,117 @@ public static boolean isEmptyPayloadStream(InputStream inputStream) throws IOExc
 
         return isEmptyPayload;
     }
+
+    /**
```
Comment (isudana) [inline components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]: Refer to https://www.oracle.com/technetwork/java/javase/tech/index-137868.html to learn about how to write javadocs.

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java (hunk 1 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java (hunk 2 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java (hunk 3 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/Axis2HttpTransportListener.java [hunk 9]
```diff
@@ -36,50 +37,79 @@
...
 
+    private ServerConnector serverConnector;
+    private WorkerPool workerPool;
+    private ConfigurationContext configurationContext;
+
     @Override
     public void init(ConfigurationContext configurationContext, TransportInDescription transportInDescription) {
 
-        WorkerPool workerPool = WorkerPoolFactory.getWorkerPool(BridgeConstants.DEFAULT_WORKER_POOL_SIZE_CORE,
+        LOG.info("Init Axis2HttpTransportListener");
```
[the remark this answers is not in the document]
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/Axis2HttpTransportListener.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/Axis2HttpTransportListener.java [hunk 10]
```diff
@@ -36,50 +37,79 @@
...
                 BridgeConstants.DEFAULT_WORKER_POOL_SIZE_MAX,
                 BridgeConstants.DEFAULT_WORKER_THREAD_KEEPALIVE_SEC,
                 BridgeConstants.DEFAULT_WORKER_POOL_QUEUE_LENGTH,
                 BridgeConstants.HTTP_WORKER_THREAD_GROUP_NAME,
                 BridgeConstants.HTTP_WORKER_THREAD_ID);
 
-        HttpWsConnectorFactory httpWsConnectorFactory = new DefaultHttpWsConnectorFactory();
+        int portOffset = Integer.parseInt(System.getProperty("portOffset", "0"));
+        Parameter portParam = transportInDescription.getParameter(TransportListener.PARAM_PORT);
+        int port = Integer.parseInt(portParam.getValue().toString());
```
[the remark this answers is not in the document]
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/Axis2HttpTransportListener.java]: Fixed. Will throw an exception now.

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java [hunk 11]
```diff
@@ -67,12 +68,15 @@
 
     @Override
     public void init(ConfigurationContext configurationContext, TransportOutDescription transportOutDescription) {
+
+        LOG.info("init");
```
[the remark this answers is not in the document]
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/RequestUtils.java (hunk 4 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/listener/RequestUtils.java]: Just a default value I guess. The user of the method just check whether the version is 1 or else. 
@menuka94 may be able to provide an explanation.

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java (hunk 5 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java (hunk 6 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java (hunk 7 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java]: done

File: components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java (hunk 8 above)
Comment (tharinduwijewardane) [reply components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/util/MessageUtils.java]: Will do.

**Pull request merged**